src/core/geometry/qgslinestring.cpp

```diff
@@ -187,6 +187,90 @@ bool QgsLineString::isEmpty() const
   return mX.isEmpty();
 }
 
+QgsLineString *QgsLineString::asGridified( double hSpacing, double vSpacing, double dSpacing, double mSpacing,
+    double /*tolerance*/, SegmentationToleranceType /*toleranceType*/ ) const
+{
+  int length = numPoints();
+
+  if ( length <= 0 )
+    return nullptr;
+
+
+  QgsLineString *result;
+
+
+  // helper functions
+  auto roundVertex = [&]( QgsPoint & out, int i )
+  {
+    if ( hSpacing > 0 )
+      out.setX( round( mX.at( i ) / hSpacing ) * hSpacing );
```

Comment:
I think (but aren't certain) that this needs to be `std::round` for msvc to be happy.

Reply:
I'll check but I think I saw it like this somewhere else and I just copied.

Reply:
That was probably before recent changes (made last week) to swap all the rounds to std::round and avoid qRound

Reply:
It was in the hackfest so yes ;)
I'll change it.